rasa/core/agent.py

```diff
@@ -389,8 +389,14 @@ def update_model(
 
         # update domain on all instances
         self.tracker_store.domain = domain
+        if hasattr(self.nlg, "responses"):
+            self.nlg.responses = domain.responses if domain else {}
+
         if hasattr(self.nlg, "templates"):
-            self.nlg.templates = domain.templates if domain else {}
+            rasa.shared.utils.io.raise_deprecation_warning(
+                "Please use the `responses` attribute instead of the `templates` attribute to manage responses. Please see https://rasa.com/docs/rasa/migration-guide/ for more detailed instructions."
```

Comment:
How about linking to the specific section of the migration guide? `raise_deprecating_warning` also has a separate parameter `docs` for the documentation link.

Reply:
Thanks for the tip!